pkcs11/src/new/functions/decryption.rs

```diff
@@ -1 +1,5 @@
-// to add: decrypt_init, decrypt
+impl Pkcs11 {
+    pub fn decrypt(&self, session: &SessionHandle, mechanism: Mechanism, key: ObjectHandle, encrypted_data: Vec<u8>) -> Result<Vec<u8>> {
```

Comment:
Add a doc comment that those are single-parts. See if putting `&mut` help with multi-threading.
See multi-threading problems for `ObjectHandle`